Scanning crashes on Windows phone on device rotation, this fixes #213

daserge opened the pull request:
Fixes scanning in portrait orientation, this fixes #172, this fixes #202

vladimir-kotikov commented on src/windows/BarcodeScannerProxy.js:
```diff
@@ -243,7 +246,19 @@ module.exports = {
+            try {
+                return capture.videoDeviceController.setMediaStreamPropertiesAsync(Windows.Media.Capture.MediaStreamType.videoPreview, videoEncodingProperties);
+            } catch (e) {
+                if (e.number !== PHOTO_SEQUENCE_IS_IN_PROGRESS) {
```
Could you add a comment clarifying why we can safely ignore this error?

daserge replied:
This error is not documented anywhere, I can just assume that we can't apply MediaStreamProperties while a photo sequence is in progress - but the error constant name says this for itself.

vladimir-kotikov commented on src/windows/BarcodeScannerProxy.js:
```diff
@@ -243,7 +246,19 @@ module.exports = {
+                return capture.videoDeviceController.setMediaStreamPropertiesAsync(Windows.Media.Capture.MediaStreamType.videoPreview, videoEncodingProperties);
+            } catch (e) {
+                if (e.number !== PHOTO_SEQUENCE_IS_IN_PROGRESS) {
+                    throw e;
```
Since `updatePreviewForRotation` is returning promise, wouldn't it be better to reject the resultant promise rather that throw synchronous exception?

daserge replied:
There is no intermediate promise variable - so could you please explain how can we reject a returned promise?
I believe it's ok to throw an exception as it will be caught in the topmost promise chain' done.

vladimir-kotikov replied:
> ok to throw an exception as it will be caught in the topmost promise

Yeah, i noticed that. It is more like a good pattern to follow - function should return both success and error consistently - either in sync or async way.

vladimir-kotikov commented on src/windows/BarcodeScannerProxy.js:
```diff
@@ -243,7 +246,19 @@ module.exports = {
+            } catch (e) {
+                if (e.number !== PHOTO_SEQUENCE_IS_IN_PROGRESS) {
+                    throw e;
+                } else { // Retry:
```
nit: `else` statement is excess here

vladimir-kotikov commented:
@daserge, could you please also rebase this

daserge commented:
Reworked and rebased, @vladimir-kotikov could you please take a look once more?
Scanning in portrait mode now works although the preview is being shrinked in this orientation.

vladimir-kotikov commented:
Thanks, @daserge. I'll verify this on my devices and merge if everything is good

vladimir-kotikov commented:
Tested on Lumia 730 w/ Windows 10 and Lumia 822 - everything seems to be working
